processor/src/main/java/org/mapstruct/ap/internal/model/Javadoc.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.internal.model;
+
+import org.mapstruct.ap.internal.model.common.ModelElement;
+import org.mapstruct.ap.internal.model.common.Type;
+import org.mapstruct.ap.internal.util.Strings;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class Javadoc extends ModelElement {
+
+    public static class Builder {
+
+        private String value;
+        private List<String> authors;
+        private String deprecated;
+        private String since;
+
+        public Builder value(String value) {
+            this.value = value;
+            return this;
+        }
+
+        public Builder authors(List authors) {
+            this.authors = authors;
+            return this;
+        }
+
+        public Builder deprecated(String deprecated) {
+            this.deprecated = deprecated;
+            return this;
+        }
+
+        public Builder since(String since) {
+            this.since = since;
+            return this;
+        }
+
+        public Javadoc build() {
+            return new Javadoc(
+                    value,
+                    authors,
+                    deprecated,
+                    since
+            );
+        }
+    }
+
+    private final String value;
+    private final List<String> authors;
+    private final String deprecated;
+    private final String since;
+
+    private Javadoc(String value, List<String> authors, String deprecated, String since) {
+        this.value = value;
+        this.authors = authors;
```

Comment:
We can do the defensive part of the authors here. i.e. `this.authors = authors != null ? Collections.unmodifiableList( authors ) : Collections.emptyList()`.

Doing this would simplify both the `hasAuthors` and `getAuthors`

Reply:
I agree with with.